### PR TITLE
Pr

### PR DESCRIPTION
**Pull request by niharikadayama**

### CODE REVIEW

File: src/redux/shopping/reducer.tsx [hunk 1]
```diff
@@ -1,104 +1,33 @@
...
-            return {
-               ...state,
-               cart: inCart
-                ? state.cart.map((item:any) => 
-                   item.id === action.payload.id 
-                    ? {...item, qty: item.qty + 1} 
-                    : item
-                )
-                : [...state.cart,{...item, qty: 1}]
+      case actionTypes.ADD_TO_CART:
```
Comment (Parth-coditas) [inline src/redux/shopping/reducer.tsx:14]: You can refactor this case as well.

```
 const item = materials.find(prod => prod.id === action.payload.id) 
        // check if item is in cart already
        const inCart = state.cart.find((item:any)=> 
        item.id === action.payload.id ? true : false
        );
```
no need to loop twice and also return true or false
as item.id === action.payload itself is a condition which will directly return true or false.

File: src/screen/itemDetail/index.tsx [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 import styles from "./styles";
 import Icon from 'react-native-vector-icons/FontAwesome'
 import {Button} from "components";
-import {colors, size} from "theme"
+import {Colors, size} from "theme"
 
-const ItemDetail = ({navigation,route, addToCart, addToWishlist}) =>{
+const ItemDetail = ({navigation,route}) =>{
     const item = route.params
     const sizeCheck = item.size
```
Comment (Parth-coditas) [inline src/screen/itemDetail/index.tsx:12]: Can you find an alternative / better approach to use sizeCheck in the FlatList ?

File: src/screen/details/index.tsx [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
 import {View, Text, Image, TouchableOpacity, FlatList} from 'react-native';
-import {connect} from 'react-redux';
 import {materials} from 'constant';
-import Icon from 'react-native-vector-icons/FontAwesome';
 import {SearchBar, Header} from 'components';
 import styles from './styles';
 
 const Details = ({navigation, route}) => {
   const item = route.params;
   const DataType = item.key;
```
Comment (Parth-coditas) [inline src/screen/details/index.tsx:9]: Find an alternative / better approach to use DataType in the FlatList

**Review by Parth-coditas — CHANGES_REQUESTED** [review]

**Comment by Parth-coditas** [comment]
Also, 
keep branch names, commit messages more meaningful to explain what functionality it has, or what changes did that commit had !

File: package.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
     "react-native-safe-area-context": "^3.3.2",
     "react-native-screens": "^3.8.0",
     "react-native-vector-icons": "^8.1.0",
+    "react-navigation": "^4.4.4",
```
[the remark this answers is not in the document]
Comment (niharikadayama) [reply package.json:27]: installed it to use switch navigator .. using createSwitchNavigator from 'react-navigation'

[the remark this answers is not in the document]
Comment (Parth-coditas) [reply package.json:27]: Switch Navigator is mostly deprecated, and you are already using react-navigation v6, so no need to install react-navigation v4 again, instead you can find an alternative to switch navigator

[the remark this answers is not in the document]
Comment (niharikadayama) [reply package.json:27]: okay got it. removed v4.

File: src/navigation/stackNavigation.tsx [hunk 5]
```diff
@@ -1,28 +1,39 @@
...
-    <Stack.Navigator screenOptions={{header: () => null}}>
-      <Stack.Screen name="splash" component={Splash} />
-      <Stack.Screen name="login" component={Login} />
-      <Stack.Screen name="signup" component={SignUp} />
-      <Stack.Screen name="root" component={BottomTabNavigator} />
-      <Stack.Screen name="details" component={Details} />
-      <Stack.Screen name="cart" component={Cart} />
-      <Stack.Screen name="ItemDetail" component={ItemDetail} />
+    <Stack.Navigator>
+      {true ? (
```
[the remark this answers is not in the document]
Comment (niharikadayama) [reply src/navigation/stackNavigation.tsx:22]: I was just testing something its not a part of code.

File: src/screen/settings/index.tsx [hunk 6]
```diff
@@ -1,24 +1,28 @@
...
-import {View, Text, TouchableOpacity} from 'react-native';
+import {View, Text, TouchableOpacity,Switch} from 'react-native';
 import styles from './styles';
+import Header from 'components/header';
 import Icon from 'react-native-vector-icons/Ionicons';
-import ToggleSwitch from 'toggle-switch-react-native';
 import {Colors} from 'theme';
 
-const Settings = () => {
+const Settings = ({navigation}) => {
```
Comment (Parth-coditas) [inline src/screen/settings/index.tsx:8]: Again..Redundant code
We have navigation-service file, right ?

File: src/screen/support/index.tsx [hunk 7]
```diff
@@ -23,49 +23,54 @@ const Support = () => {
...
+                              <Text style={style.cardHeading}>{item.heading}</Text>
+                          </View>
+                          <Icon
+                            name="chevron-right"
+                            size={25}
+                            style={{position: 'absolute', right: 10, top: 25}}
+                          />
+                          {index === currentIndex && (
+                              <View style={style.subHeadinglist}>
+                                  {item.subHeading.map(subHeading => (
```
Comment (Parth-coditas) [inline src/screen/support/index.tsx:60]: Any specific reason to use map ? instead of flatList ?

File: src/screen/itemDetail/index.tsx [hunk 8]
```diff
@@ -35,65 +35,63 @@ const ItemDetail = ({navigation, route}) => {
...
+                <Text style={styles.detailInfo}>{item.info}</Text>
 
-          <View style={styles.detailBottom}>
-            <Text style={styles.detailPrice}>Rs. {item.price}</Text>
+                <Icon
+                  name="heart"
+                  size={25}
+                  style={styles.detailHeart}
+                  onPress={() => {
+                    navigation.navigate('wishlist', item),
```
Comment (Parth-coditas) [inline src/screen/itemDetail/index.tsx:60]: use semicolon to separate lines, instead of commas

**Review by Parth-coditas — CHANGES_REQUESTED** [review]

**Pull request merged**